chain/yoda/execute.go

```diff
@@ -50,17 +53,37 @@ func SubmitReport(c *Context, l *Logger, id otypes.RequestID, reps []otypes.RawR
 		l.Error(":exploding_head: Failed to build tx with error: %s", err.Error())
 		return
 	}
-
-	res, err := cliCtx.BroadcastTxCommit(out)
-	if err != nil {
+	var res sdk.TxResponse
+	found := false
+	for try := 1; try <= MaxTry; try++ {
+		l.Info("Try to broadcast: %d/%d", try, MaxTry)
+		res, err = cliCtx.BroadcastTxSync(out)
+		if err == nil {
+			found = true
+			break
+		}
 		l.Error(":exploding_head: Failed to broadcast tx with error: %s", err.Error())
+		time.Sleep(SleepTime)
+	}
+	if !found {
+		l.Error(":exploding_head: Can't try to broadcast more than %d try", MaxTry)
 		return
 	}
-	if res.Code != 0 {
-		l.Error(":exploding_head: Tx returned nonzero code %d with log %s, tx hash: %s", res.Code, res.RawLog, res.TxHash)
+	for start := time.Now(); time.Since(start) < c.broadcastTimeout; {
+		time.Sleep(SleepTime)
+		txRes, err := utils.QueryTx(cliCtx, res.TxHash)
+		if err != nil {
+			l.Debug("Failed to query tx with error: %s", err.Error())
+			continue
+		}
+		if txRes.Code != 0 {
+			l.Error(":exploding_head: Tx returned nonzero code %d with log %s, tx hash: %s", txRes.Code, txRes.RawLog, txRes.TxHash)
+			return
+		}
+		l.Info(":smiling_face_with_sunglasses: Successfully broadcast tx with hash: %s", txRes.TxHash)
 		return
 	}
-	l.Info(":smiling_face_with_sunglasses: Successfully broadcast tx with hash: %s", res.TxHash)
+	l.Info("Can't find tx hash: %s before timeout", res.TxHash)
```

Comment:
❓ Cannot get transaction response from hash: %s transaction might be included in the next few blocks or check your node's health.